pallets/lbp/src/tests.rs

```diff
@@ -0,0 +1,1680 @@
+				ALICE,
+				LBPAssetInfo {
+					id: HDX,
+					amount: 1_000_000_000,
+					initial_weight: 20,
+					final_weight: 90,
+				},
+				LBPAssetInfo {
+					id: DOT,
+					amount: 2_000_000_000,
+					initial_weight: 80,
+					final_weight: 10,
+				},
+				(10u64, 20u64),
+				WeightCurveType::Linear,
+				true,
+			),
+			BadOrigin
+		);
+	});
+}
+
+#[test]
+fn create_same_pool_should_not_work() {
+	new_test_ext().execute_with(|| {
+		assert_ok!(LBPPallet::create_pool(
+			Origin::root(),
+			ALICE,
+			LBPAssetInfo {
+				id: ACA,
+				amount: 1_000_000_000,
+				initial_weight: 20,
+				final_weight: 90,
+			},
+			LBPAssetInfo {
+				id: DOT,
+				amount: 2_000_000_000,
+				initial_weight: 80,
+				final_weight: 10,
+			},
+			(10u64, 20u64),
+			WeightCurveType::Linear,
+			true,
+		));
+
+		assert_noop!(
+			LBPPallet::create_pool(
+				Origin::root(),
+				ALICE,
+				LBPAssetInfo {
+					id: ACA,
+					amount: 10_000_000_000,
+					initial_weight: 30,
+					final_weight: 70,
+				},
+				LBPAssetInfo {
+					id: DOT,
+					amount: 20_000_000_000,
+					initial_weight: 70,
+					final_weight: 30,
+				},
+				(100u64, 200u64),
+				WeightCurveType::Linear,
+				true,
+			),
+			Error::<Test>::PoolAlreadyExists
+		);
+
+		expect_events(vec![
+			Event::PoolCreated(ALICE, ACA, DOT, 1_000_000_000, 2_000_000_000).into()
+		]);
+	});
+}
+
+#[test]
+fn create_pool_with_same_assets_should_not_work() {
+	new_test_ext().execute_with(|| {
+		assert_noop!(
+			LBPPallet::create_pool(
+				Origin::root(),
+				ALICE,
+				LBPAssetInfo {
+					id: ACA,
+					amount: 1_000_000_000,
+					initial_weight: 20,
+					final_weight: 90,
+				},
+				LBPAssetInfo {
+					id: ACA,
+					amount: 2_000_000_000,
+					initial_weight: 80,
+					final_weight: 10,
+				},
+				(20u64, 10u64),
+				WeightCurveType::Linear,
+				true,
+			),
+			Error::<Test>::CannotCreatePoolWithSameAssets
+		);
+	});
+}
+
+#[test]
+fn create_pool_invalid_data_should_not_work() {
+	new_test_ext().execute_with(|| {
+		assert_noop!(
+			LBPPallet::create_pool(
+				Origin::root(),
+				ALICE,
+				LBPAssetInfo {
+					id: ACA,
+					amount: 1_000_000_000,
+					initial_weight: 20,
+					final_weight: 90,
+				},
+				LBPAssetInfo {
+					id: DOT,
+					amount: 2_000_000_000,
+					initial_weight: 80,
+					final_weight: 10,
+				},
+				(20u64, 10u64), // reversed interval, the end precedes the beginning
+				WeightCurveType::Linear,
+				true,
+			),
+			Error::<Test>::InvalidBlockNumber
+		);
+	});
+}
+
+#[test]
+fn update_pool_data_should_work() {
+	predefined_test_ext().execute_with(|| {
+		// update starting block and final weights
+		assert_ok!(LBPPallet::update_pool_data(
+			Origin::signed(ALICE),
+			ACA_DOT_POOL_ID,
+			Some(15),
+			None,
+			Some(((ACA, 10), (DOT, 90))),
+			None,
+		));
+
+		// verify changes
+		let updated_pool_data = LBPPallet::pool_data(ACA_DOT_POOL_ID);
+		assert_eq!(updated_pool_data.start, 15);
+		assert_eq!(updated_pool_data.end, 20);
+
```

Comment:
the update of initial weights and final weights is not tested. 

i would also add a simple test to verify that if you update one ting (eg. start) - the rest pool parameters are preserved and not changed.